packages/drivers/odsp-driver/src/odspUtils.ts

```diff
@@ -198,8 +199,8 @@ export async function fetchAndParseAsJSONHelper<T>(
         throwOdspNetworkError(
             "errorWhileParsingFetchResponse",
             fetchIncorrectResponse,
-            content,
-            undefined,
+            content, // response
+            text,
             { error: Object(e) },
```

Comment:
This is because e.message is known to contain TMI, per comment above?  Can you move the comment down here?  And consider using wrapError instead.  You give it a callback that takes the message, you can put it in a tagged property.

In other words (pseudocode off the top of my head):

```
} catch(e) {
    throw wrapError(e, (errMsg) => createOdspNetworkError(..., { jsonParseErrorMessage: { tag: UserData, value: errMsg }}));
}
```

Reply:
I do not want to touch IOdspResponse<> in this PR, that would make it much bigger (re rename of content).
I believe we are fine passing respose text down. It get recorded to telemetry only if it passes tryParseErrorResponse() check - see createOdspNetworkError().

Note this is not an object - text is expected, so no way to wrap anything here.

As for comment, it refers to error object itself, not text of response. Logic is not changed RE how we handle it here.
I believe we currently drop "error" property as it's complex object that is not tagged property. It's a bit risky to rely on that, so I'll remove it here.

Reply:
> I do not want to touch IOdspResponse<> in this PR

Wasn't proposing this, not sure why you mention it 🤷‍♂️

> I believe we are fine passing respose text down

Agreed.

> text is expected, so no way to wrap anything here

This is one feature of `wrapError` - it doesn't care what you give it, it will "do the right thing" and pass a string `errorMessage` to callback.